backend/src/main/java/com/capstone/moneytree/service/impl/DefaultTransactionService.java

```diff
@@ -127,12 +127,46 @@ private void executeTransaction(String alpacaKey, Order order, User user) {
    }
 
    private Transaction constructTransactionFromOrder(Order alpacaOrder) {
-      return Transaction.builder().status(TransactionStatus.PENDING).purchasedAt(alpacaOrder.getCreatedAt())
+      return Transaction.builder()
+              .status(TransactionStatus.PENDING)
+              .purchasedAt(alpacaOrder.getCreatedAt())
               .clientOrderId(alpacaOrder.getClientOrderId())
               .moneyTreeOrderType(MoneyTreeOrderType
                       .valueOf(alpacaOrder.getType().toUpperCase() + "_" + alpacaOrder.getSide().toUpperCase()))
               .quantity(Float.parseFloat(alpacaOrder.getQty())).purchasedAt(alpacaOrder.getSubmittedAt())
-              .symbol(alpacaOrder.getSymbol()).build(); // avg price and total will be set only if stock got fulfilled
+              .symbol(alpacaOrder.getSymbol())
+              .build(); // avg price and total will be set only if stock got fulfilled
+   }
+
+   // ISSUE-346
+   public void updateUserScore(Order order, User user, Transaction transaction) {
+      if (order.getSide().equals("sell")) {
+         float boughtPrice = 0;
+         List<Made> userMade = madeDao.findByUserId(user.getId());
+         for (Made made : userMade) {
+            Transaction oldTransaction = made.getTransaction();
+            if (oldTransaction.getSymbol().equals(transaction.getSymbol())) {
```

Comment:
If the user bought AAPL stocks in different transactions (different days and different quantities and prices), then we will be calculating his score based on his last transaction only. I think maybe we can calculate the average purchase price based on quantity, and then calculate the price with that number. for example, if the user has 3 transactions with this info:
AAPL quantity: 5, price = 10 
AAPL quantity: 3, price = 20
AAPL quantity: 1, price = 5  
Average price = (10*5 + 20*3 +1*5) / (5+3+1)   = 12.78

Reply:
I know that Alessandro wanted something really simple because of time constraints for this project and features that we have to pull out. I also know that this only checks the last transaction. I had in mind before starting this feature to do something like you mentioned with stock quantity purchased. However, Alessandro told me not to worry about it. So I'm not sure what's the idea here anymore.

@Alessandro100 what do you think?

Reply:
Edit: Actually I think you're right Hossein. Your solution is better. I'll work on it.

Reply:
Just pushed some new changes. I followed that formula since I think it is better if the user has more than one transaction done.

Reply:
Very nice David, thanks for making the changes and for changing the tests as well. I approved this PR.